tools/backups2datalad/datasetter.py

```diff
@@ -249,14 +243,19 @@ async def get_dandisets(
                 else:
                     yield d
 
-    async def set_superds_description(
-        self, superds: AsyncDataset, ds_stats: list[DatasetStats]
-    ) -> None:
+    async def set_superds_description(self, superds: AsyncDataset) -> None:
         log.info("Setting repository description for superdataset")
         repo = await superds.get_ghrepo()
-        total_size = naturalsize(sum(s.size for s in ds_stats))
+        qty = 0
+        size = 0
+        for s in await superds.get_subdatasets():
```

Comment:
this would not work for e.g. 000108 where we have zarrs uninstalled (and available only among dandizarrs folder) if this function is used there as well, and it seems it does.  Do you see an easy workaround?

Reply:
* The code in question is only supposed to count immediate subdatasets of the superdataset, i.e., the Dandiset datasets.  Does it do that or not?
* Are you sure that `Dataset.subdatasets()` (which this method calls) doesn't count uninstalled Datasets?  [This test works](https://github.com/dandi/dandisets/blob/861e239af7095fd44312077a3fc0941d4d68be25/tools/test_backups2datalad/test_zarr.py#L80) despite operating on a dataset with uninstalled submodules, and at one point [we call the method with `state="present"`](https://github.com/dandi/dandisets/blob/861e239af7095fd44312077a3fc0941d4d68be25/tools/backups2datalad/adataset.py#L498), implying that it doesn't only operate on installed subdatasets by default.

Reply:
> The code in question is only supposed to count immediate subdatasets of the superdataset, i.e., the Dandiset datasets. Does it do that or not?

it does

> Are you sure that Dataset.subdatasets() (which this method calls) doesn't count uninstalled Datasets?

it "counts" (as in number) but cannot possibly get their stats if they are not installed, can it?  looking at hte code

```python
    async def get_stored_stats(self) -> Optional[DatasetStats]:
        if (stored_stats := self.ds.config.get("dandi.stats", None)) is not None:
            try:
                stored_commit, files_str, size_str = stored_stats.split(",")
                files = int(files_str)
                size = int(size_str)
            except Exception:
                return None
            if stored_commit == await self.get_commit_hash():
                return DatasetStats(files=files, size=size)
        return None
```
it would just return `None` and thus size would not be counted properly.

Reply:
The code for getting the stats for a Dandiset dataset maps Zarr subdatasets to their clones in `dandizarrs/` [here-ish](https://github.com/dandi/dandisets/blob/df2b83c60d9f1cfd966aecd8dade9c231d8a7e7f/tools/backups2datalad/adataset.py#L409-L419), so it should all work.

Reply:
oh, cool, sorry I missed that. Let's give it a shot then!